thriftbp/client_pool.go

```diff
@@ -169,6 +169,22 @@ type ClientPoolConfig struct {
 	BreakerConfig *breakerbp.Config
 }
 
+// Validate checks the ClientPoolConfig for any missing or erroneous values. It
+// returns an errorsbp.Batch to be combineable with other error batching.
+func (c ClientPoolConfig) Validate() error {
+	var batch errorsbp.Batch
+	if c.ServiceSlug == "" {
+		batch.Add(errors.New("ServiceSlug cannot be empty"))
```

Comment:
Can you also use named errors for these rather than using errors.New each time?

Reply:
+1, and also linter will yell at you when you try to capitalize error messages (but that seems to only check `fmt.Errorf` but not `errors.New`).

Reply:
For some reason I thought `batch.Compile()` produces a new error

Reply:
Does that mean we want to lowercase those strings, i.e.
```
serviceSlug cannot be empty
```

Reply:
nevermind I just realized that it's capitalized because it's referring to the field name that's already capitalized. I think just leave them as they are now is fine.